docs/testing-operators.md

```diff
@@ -1,120 +1,404 @@
+
+```
+kubectl get catalogsourceconfig -n marketplace
+
+NAME                      STATUS      MESSAGE                                       AGE
+johndoe-operators         Succeeded   The object has been successfully reconciled   93s
+```
+
+Additionally, a `CatalogSource` is created in the namespace indicated in `spec.targetNamespace` (in the above example, `operators`):
+
+```
+kubectl get catalogsource -n olm
+
+NAME                           NAME        TYPE   PUBLISHER   AGE
+johndoe-operators              Custom      grpc   Custom      3m32s
+[...]
+```
+### 7. Create an OperatorGroup
+
+An `OperatorGroup` is used to denote which namespaces your Operator should be watching. It must in the namespace where your operator should be deployed, we'll use `default` in this example:
+
+> If your Operator supports watching all namespaces (as indicated by `spec.installModes` in the CSV) you can omit the following step and place your `Subscription` in the `operators` namespace instead.
+
+
+Create the following as the file `operator-group.yaml`
+
+```
+apiVersion: operators.coreos.com/v1alpha2
+kind: OperatorGroup
+metadata:
+  name: my-operatorgroup
+  namespace: default
+spec:
+  targetNamespaces:
+  - default
+```
+
+Deploy the `OperatorGroup` resource:
+
+```
+kubectl apply -f operator-group.yaml
+```
+
+### 8. Create a Subscription
+
+The last piece ties together all of the previous steps. A `Subscription` is created to the operator. Save the following to a file named: `operator-subscription.yaml`:
+
+```
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: my-operator-subsription
+  namespace: default
+spec:
+  channel: <channel-name>
+  name: my-operator
+  source: johndoe-operators
+  sourceNamespace: olm
+```
+
+### 9. Verify Operator health
+
+Watch your Operator being deployed by OLM from the catalog source created by Operator Marketplace with the following command:
+
+```
+kubectl get clusterserviceversion -n default
+
+NAME                 DISPLAY       VERSION   REPLACES   PHASE
+my-operator.v1.0.0   My Operator   1.0.0                Succeeded
+```
+
+> The above command assumes you have created the `Subscription` in the `default` namespace. Adjust accordingly if you have selected a different namespace.
+
+
+If your Operator deployment (CSV) shows a `Succeeded` in the `InstallPhase` status, your Operator is deployed successfully. If that's not the case check the `ClusterServiceVersion` objects status for details.
+
+Optional also check your Operator's deployment:
+
+```
+kubectl get deployment -n default
+```
+
+## Testing Operator Deployment on OpenShift
+
+On OpenShift Container Platform and OKD 4.1 or newer `operator-marketplace` and `operator-lifeycle-manager` are already installed. You can start right away by creating an `OperatorSource` in the `openshift-marketplace` namespace as a user with the `cluster-admin` role.
+
+### 1. Create the OperatorSource
```

Comment:
I suggest making this common across minikube and OpenShift otherwise you will have to maintain this in two spots.

Reply:
It's repetitive, but we would need to break up the sections covering minikube and OpenShift to avoid it. The logical flow and isolation of these two sections would broken just to save some text.